caikit/runtime/service_generation/data_stream_source.py

```diff
@@ -263,14 +281,14 @@ def make_data_stream_source(data_element_type: Type) -> Type[DataBase]:
             data_object,
         )
         setattr(
-            sys.modules[__name__],
+            sys.modules[data_object.__module__],
             cls_name,
             data_object,
         )
 
         # Add an init that sequences the initialization so that
         # DataStreamSourceBase is initialized after DataBase
-        orig_init = data_object.__init__
+        orig_init = _DataObjectBaseMetaClass._make_init(data_object.fields)
```

Comment:
This is needed because the logic to detect whether or not there's a "custom __init__" fails since the class inherits from `DataStreamSourceBase`. Rather than make that logic more complicated, I added this hack.